### PR TITLE
Keep only ruleID instead of the full Rucio rule.

### LINKED ISSUE #10077: Keep only ruleID instead of the full Rucio rule for MSRuleCleaner
**Impact of the bug**
MSRuleCleaner

**Describe the bug**
NOT exactly a bug:
While validating the Service in testbed, we figured out that we have quite a big and extensive log we preserve per every workflow. And while running the service in DEBUG mode at some point we are dumping the whole workflow representation into the log. Which is not a bug by itself, but we are also keeping the full Rucio rule in the workflow, which contains a lot of unneeded information like [1]. And this is just for a DID with only two blocks, while we are having some containers with  several K blocks. The only reason for recording all that information into the workflow was just for keeping into the log the `ruleID`s we are deleting associated with the `DID` (block or container) to which it relates. And since now we know the information (the link between `ruleID` and `DID`) is preserved into the Rucio database (even though not that easy to be accessed), we decided to drop the` DID` and go only with the `ruleID`.

We should also change the log level for the deletion message itself from DEBUG to INFO.

[1]
```
 'RulesToClean': {'plineAgentBlock': [{u'account': u'wma_test',
                                       u'activity': u'Production Output',
                                       u'child_rule_id': None,
                                       u'comments': u'WMAgent automatic container rule',
                                       u'copies': 1,
                                       u'created_at': datetime.datetime(2020, 11, 17, 14, 32, 49),
                                       u'did_type': u'DATASET',
                                       u'eol_at': None,
                                       u'error': None,
                                       u'expires_at': datetime.datetime(2020, 11, 17, 20, 50, 30),
                                       u'grouping': u'DATASET',
                                       u'id': u'00f877d0f91241d9a24be04cebf10f93',
                                       u'ignore_account_limit': False,
                                       u'ignore_availability': False,
                                       u'locked': False,
                                       u'locks_ok_cnt': 1,
                                       u'locks_replicating_cnt': 0,
                                       u'locks_stuck_cnt': 0,
                                       u'meta': u'"{\\"agentHost\\": \\"vocms0193.cern.ch\\", \\"userAgent\\": \\"WMAgent\\"}"',
                                       u'name': u'/RelValDarkSUSY_14TeV/DMWM_Test-SC_LHE_Ext_HG2011_Val_Todor_MSRuleCleaner_v1-v11/GEN-SIM#b9a66f7e-27a6-4f00-9b61-4ea431473dcb',
                                       u'notification': u'NO',
                                       u'priority': 3,
                                       u'purge_replicas': False,
                                       u'rse_expression': u'T2_CH_CERN_Test',
                                       u'scope': u'cms',
                                       u'source_replica_expression': None,
                                       u'split_container': False,
                                       u'state': u'OK',
                                       u'stuck_at': None,
                                       u'subscription_id': None,
                                       u'updated_at': datetime.datetime(2020, 11, 17, 19, 50, 30),
                                       u'weight': None},
                                      {u'account': u'wma_test',
                                       u'activity': u'Production Output',
                                       u'child_rule_id': None,
                                       u'comments': u'WMAgent automatic container rule',
                                       u'copies': 1,
                                       u'created_at': datetime.datetime(2020, 11, 17, 14, 42, 51),
                                       u'did_type': u'DATASET',
                                       u'eol_at': None,
                                       u'error': None,
                                       u'expires_at': datetime.datetime(2020, 11, 17, 20, 50, 30),
                                       u'grouping': u'DATASET',
                                       u'id': u'adf1fda1a29c4cc3b15d52ad4e5af4bc',
                                       u'ignore_account_limit': False,
                                       u'ignore_availability': False,
                                       u'locked': False,
                                       u'locks_ok_cnt': 7,
                                       u'locks_replicating_cnt': 0,
                                       u'locks_stuck_cnt': 0,
                                       u'meta': u'"{\\"agentHost\\": \\"vocms0193.cern.ch\\", \\"userAgent\\": \\"WMAgent\\"}"',
                                       u'name': u'/RelValDarkSUSY_14TeV/DMWM_Test-SC_LHE_Ext_HG2011_Val_Todor_MSRuleCleaner_v1-v11/GEN-SIM-DIGI-RAW#2655420e-c4ea-401f-ba2d-c7969a3ad92b',
                                       u'notification': u'NO',
                                       u'priority': 3,
                                       u'purge_replicas': False,
                                       u'rse_expression': u'T2_CH_CERN_Test',
                                       u'scope': u'cms',
                                       u'source_replica_expression': None,
                                       u'split_container': False,
                                       u'state': u'OK',
                                       u'stuck_at': None,
                                       u'subscription_id': None,
                                       u'updated_at': datetime.datetime(2020, 11, 17, 19, 50, 30),
                                       u'weight': None}],
                  'plineAgentCont': []},
```

**How to reproduce it**
Just read the logs of the current implementation of the service

**Expected behavior**
Just keep the `ruleID` into the workflow representation and log only it while deleting.

**Additional context and error message**
Add any other context about the problem here, like error message and/or traceback. You might want to use triple back ticks to properly format it.

### PR DESCRIPTION
**Pull request by todor-ivanov**
Fixes #10077 

#### Status
ready

#### Description
With this change we drop the full Rucio rule that we were keeping  in the workflow representation, and we are just preserving the ruleID. 

#### Is it backward compatible (if not, which system it affects?)
YES

#### Related PRs
NO

#### External dependencies / deployment changes
NO

### CODE REVIEW

File: src/python/WMCore/MicroService/Unified/MSRuleCleaner.py [hunk 1]
```diff
@@ -396,12 +396,16 @@ def getRucioRules(self, wflow, gran, rucioAcct):
         # Find all the output placement rules created by the agents
         for dataCont in wflow['OutputDatasets']:
             if gran == 'container':
-                wflow['RulesToClean'][currPline].extend(self.rucio.listDataRules(dataCont, account=rucioAcct))
+                ruleIds = [rule['id'] for rule in
```
Comment (amaltaro) [inline src/python/WMCore/MicroService/Unified/MSRuleCleaner.py:399]: I guess you already know I'm not a big fan of multi-lines list comprehension, so let me suggest a construction that looks clearer in my opinion. How about:
```
for rule in self.rucio.listDataRules(dataCont, account=rucioAcct):
    wflow['RulesToClean'][currPline].append(rule['id'])
```
?

If you agree, please change the block-level code (below) as well

File: src/python/WMCore/MicroService/Unified/MSRuleCleaner.py [hunk 2]
```diff
@@ -422,21 +426,18 @@ def cleanRucioRules(self, wflow):
         delResults = []
         if self.msConfig['enableRealMode']:
             for rule in wflow['RulesToClean'][currPline]:
-                msg = "%s: Deleting rule['id']: %s "
-                msg += "For did: %s"
-                self.logger.info(msg, currPline, rule['id'], rule['name'])
-                delResult = self.rucio.deleteRule(rule['id'])
+                msg = "%s: Deleting ruleId: %s "
```
Comment (amaltaro) [inline src/python/WMCore/MicroService/Unified/MSRuleCleaner.py:429]: Please compact this `msg` and the line below altogether, such as:
```
self.logger.info("%s: Deleting ruleId: %s ", currPline, rule)
```

Line is not long enough and there is no need to break it in multiple lines.

File: src/python/WMCore/MicroService/Unified/MSRuleCleaner.py [hunk 3]
```diff
@@ -422,21 +426,18 @@ def cleanRucioRules(self, wflow):
...
+                self.logger.info(msg, currPline, rule)
+                delResult = self.rucio.deleteRule(rule)
                 delResults.append(delResult)
                 if not delResult:
-                    msg = "%s: Failed to delete rule['id']: %s "
-                    msg += "For did: %s"
-                    self.logger.warning(msg, currPline, rule['id'], rule['name'])
+                    msg = "%s: Failed to delete ruleId: %s "
```
Comment (amaltaro) [inline src/python/WMCore/MicroService/Unified/MSRuleCleaner.py:434]: Same here, one-line would do.

File: src/python/WMCore/MicroService/Unified/MSRuleCleaner.py [hunk 4]
```diff
@@ -422,21 +426,18 @@ def cleanRucioRules(self, wflow):
...
+                    self.logger.warning(msg, currPline, rule)
         else:
             for rule in wflow['RulesToClean'][currPline]:
                 delResults.append(True)
-                msg = "%s: Is about to delete rule['id']: %s "
-                msg += "For did: %s"
-                self.logger.debug(msg, currPline, rule['id'], rule['name'])
+                msg = "%s: Is about to delete ruleId: %s "
```
Comment (amaltaro) [inline src/python/WMCore/MicroService/Unified/MSRuleCleaner.py:439]: And same here too.
But here, I would suggest to adopt the same logging format as we have used for the other MS, thus starting the log record with `DRY-RUN: blah blah`

**Review by amaltaro — COMMENTED** [review]
Changes look good to me Todor, however I'd like very much to have those suggestions applied to the code please.

File: src/python/WMCore/MicroService/Unified/MSRuleCleaner.py (hunk 1 above)
Comment (todor-ivanov) [reply src/python/WMCore/MicroService/Unified/MSRuleCleaner.py:399]: Done

File: src/python/WMCore/MicroService/Unified/MSRuleCleaner.py (hunk 2 above)
Comment (todor-ivanov) [reply src/python/WMCore/MicroService/Unified/MSRuleCleaner.py:429]: Done

File: src/python/WMCore/MicroService/Unified/MSRuleCleaner.py (hunk 3 above)
Comment (todor-ivanov) [reply src/python/WMCore/MicroService/Unified/MSRuleCleaner.py:434]: Done

File: src/python/WMCore/MicroService/Unified/MSRuleCleaner.py (hunk 4 above)
Comment (todor-ivanov) [reply src/python/WMCore/MicroService/Unified/MSRuleCleaner.py:439]: Done

**Comment by todor-ivanov** [comment]
Thanks @amaltaro . I just applied the changes you requested. Please take a look again.

**Review by amaltaro — APPROVED** [review]
Thanks Todor. While we wait for Imran to deploy it in testbed, we might ask him (over mattermost) to apply this patch and buy you some extra time to validate things.

**Pull request merged**